apps/examples/src/examples/canvas-events/CanvasEventsExample.tsx

```diff
@@ -5,12 +5,28 @@ import { useCallback, useState } from 'react'
 // There's a guide at the bottom of this file!
 
 export default function CanvasEventsExample() {
+	// [1]
 	const [events, setEvents] = useState<string[]>([])
-
+	// [2]
 	const handleEvent = useCallback((data: TLEventInfo) => {
-		setEvents((events) => [JSON.stringify(data, null, 2), ...events.slice(0, 100)])
+		setEvents((currentEvents) => {
+			const parsedEvents = currentEvents.map((e) => JSON.parse(e))
```

Comment:
this is a great start! already better than what we had - let's split this change into a separate PR - i think it still needs a little more work and I don't wanna hold up this larger change :)

Reply:
This leaked as well!